pkg/operator/reconcile.go

```diff
@@ -122,3 +121,49 @@ func MergeServiceForUpdate(current, updated *uns.Unstructured) error {
 
 	return nil
 }
+
+// mergeAnnotationsExcept merges the annotations between current and updated,
+// preferring those in "updated" if there's a conflict. Keys listed in exceptKeys
+// will have the conflict order reversed; current will win.
```

Comment:
"merges the annotations" and then does what with them?

> mergeAnnotationsExcept copies all of the annotations from `current` to `updated`, except in cases where `updated` already has that annotation, unless the annotation's key appears in `exceptKeys`.

or maybe

> mergeAnnotationsExcept copies annotations from `current` to `updated` where either the annotation does not already appear in `updated`, or else the annotation's key is in `exceptKeys`.

although the naming of both the function and `exceptKeys` doesn't make a lot of sense with either of those descriptions... It's kind of a weird function...